### ISSUE TITLE
Raise warning for DEM in geographic coordinates for some terrain attributes

### ISSUE THREAD

**Issue by adehecq** [GlacioHack/xdem#634]
Some terrain attributes like slope do not make sense when the DEM is in geographic (lat/lon) coordinates. It would be great to raise a warning in such situations.
For hillshades, it is also not very good as the appearance will be very different (darker) with lat/lon. We could automatically detect this and apply the right correction factor as suggested in "gdaldem hillshade": for Meters:LatLong use scale=111120.

**Comment by rhugonnet** [comment]
Good idea! We also discussed modifying the error in `NuthKaab` for geographic coordinates into warning for #623
